src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -241,7 +239,7 @@ int IDebugPendingBreakpoint2.Bind()
 
                                     break;
                                 case enum_BP_LOCATION_TYPE.BPLT_DATA_STRING:
-                                    _address = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3);
+                                    _address = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3).Split(',')[0];
```

Comment:
Rather than repeatedly calling `HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3)` I would suggest doing:

1. Add a new internal property: `internal string AddressId { get; }`
2. Here do:

```C#
string address = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3);
if (address.Contains(","))
{
    this.AddressId = address;
    _address = address.Split(',')[0];
}
else
{
    this.AddressId = null;
    _address = address;
}
```

3. In AD7BoundBreakpoint.cs, use _pendingBreakpoint.AddressId